state/protocol/badger/mutator_test.go

```diff
@@ -2650,9 +3095,9 @@ func TestProtocolStateIdempotent(t *testing.T) {
 }
 
 func assertEpochEmergencyFallbackTriggered(t *testing.T, state realprotocol.State, expected bool) {
-	triggered, err := state.Params().EpochFallbackTriggered()
+	epochState, err := state.Final().EpochProtocolState()
```

Comment:
I think it would be helpful to generalize this to 
```suggestion
// assertEpochEmergencyFallbackTriggered tests that the given `stateSnapshot` has the `InvalidEpochTransitionAttempted` flag to the `expected` value. 
func assertEpochEmergencyFallbackTriggered(t *testing.T, stateSnapshot realprotocol.Snapshot, expected bool) {
	epochState, err := stateSnapshot.EpochProtocolState()
```